### PR TITLE
DRY up court date examples

### PR DESCRIPTION
**Pull request by kjdchapman**
@richrace and I talked about the best way to write these specs, and agreed that trying things out was the best way to tell if it made things clearer.

This PR attempts to make these specs more readable without losing the explanatory power of examples. It also does a little work on the output of those specs, attempting to express a high-level description of each case.

I'm totally happy to throw away this change if it doesn't help the readability.
_____

Old output:

image

New output:

image

### CODE REVIEW

**Comment by richrace** [comment]
I like the improvements but if we enable randomised tests (which we should but that's a different issue) the ordering will not be "pretty"

File: spec/support/shared_example_for_tenancy_classification.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@ def build_context_message(options)
   condition_matrix.each do |options|
     message = build_context_message(options)
 
-    context "when #{message}" do
+    context(options[:description] || message) do
```
Comment (richrace) [inline spec/support/shared_example_for_tenancy_classification.rb]: `message` will be generated for every `context` could we not do:

`context(options[:description] || build_context_message(options))`?

**Comment by kjdchapman** [comment]
I'm fine with the pretty ordering being broken, I think it's still clearer what's going on for a given test even with random on. 

I definitely support us turning on random order in a separate change.

Do you think it's valuable to merge just this in? I don't think doing all the classification examples at once is a good use of time.

File: spec/support/shared_example_for_tenancy_classification.rb (hunk 1 above)
Comment (kjdchapman) [reply spec/support/shared_example_for_tenancy_classification.rb]: Sure, happy to make this change (https://github.com/LBHackney-IT/lbh-income-api/pull/289/commits/58730728831437cf83b1ddda4cb0d672cd3e97b8)

**Pull request merged**